### PR TITLE
Support marking nodes for maintenance

### PR DESCRIPTION
**Pull request by tgross**
This PR provides https://github.com/joyent/containerbuddy/issues/6

The implementation is a signal handler for SIGUSR1 to mark for maintenance mode. When Containerbuddy receives SIGUSR1, it will force all polling goroutines to effectively pause by skipping their actions, and then it will fire a `FailTTL` to Consul for all the services it advertises.

I've included test rig for testing signals and a point of extension for any future signal handlers (like config reload). Per my comment in the issue, I've also changed `WriteHealthCheck` to `SendHeartbeat` for API clarity.

@misterbisson to review please.

### CODE REVIEW

File: src/containerbuddy/signals_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	handleSignals(&Config{})
+	if inMaintenanceMode() {
+		t.Errorf("Should not be in maintenance mode after starting handler")
+	}
+
+	sendAndWaitForSignal(t, syscall.SIGUSR1)
+	if !inMaintenanceMode() {
+		t.Errorf("Should be in maintenance mode after receiving SIGUSR1")
+	}
+	sendAndWaitForSignal(t, syscall.SIGUSR1)
```
Comment (misterbisson) [inline src/containerbuddy/signals_test.go]: Thanks for testing how to _unset_ maintenance mode.

**Comment by misterbisson** [comment]
This looks good to me, but I'm wondering if it's missing an update to the readme, or at least a linked ticket for updating the readme.

**Comment by tgross** [comment]
> This looks good to me, but I'm wondering if it's missing an update to the readme

Good call. Will update.

**Comment by tgross** [comment]
I've merged the other open PR and this now has a minor merge conflict. I'll fix that as well.

**Comment by tgross** [comment]
@misterbisson I've resolved the merge conflicts and updated the README.

**Comment by tgross** [comment]
Let's hold off on merging this in until I can swap out the behavior for reaping the service from Consul, per discussion in https://github.com/joyent/containerbuddy/issues/17#issuecomment-159325030

**Comment by tgross** [comment]
@misterbisson I've updated this to reap the service from Consul per the discussion in https://github.com/joyent/containerbuddy/issues/17#issuecomment-159325030

**Comment by misterbisson** [comment]
:house_with_garden: :walking:

**Comment by tgross** [comment]
Included in release: https://github.com/joyent/containerbuddy/releases/tag/0.0.2-alpha

**Pull request merged**